code/datums/memory/tattoo_kit.dm

```diff
@@ -29,38 +29,57 @@
 	qdel(ink_cart)
 	balloon_alert(tattoo_artist, "added tattoo ink")
 
-/obj/item/tattoo_kit/attack(mob/living/tattoo_holder, mob/living/tattoo_artist, params)
+/obj/item/tattoo_kit/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(.)
-		return TRUE
+	if(!proximity_flag)
+		return
+	if(!ishuman(user) || !ishuman(target))
+		return
+	var/mob/living/carbon/human/tattoo_artist = user
+	var/mob/living/carbon/human/tattoo_holder = target
 	if(!tattoo_artist.mind || tattoo_artist.combat_mode)
 		return
 	if(!uses)
-		balloon_alert(tattoo_artist, "not enough ink!")
+		tattoo_artist.balloon_alert(tattoo_artist, "not enough ink!")
 		return
-	if(!tattoo_artist.mind.memories)
-		balloon_alert(tattoo_artist, "nothing memorable to engrave!")
+	var/is_right_clicking = LAZYACCESS(params2list(click_parameters), RIGHT_CLICK)
```

Comment:
Avoid this, I plan on splitting the click chain at some point and these all need to be changed. Use `afterattack_secondary` instead.

Reply:
wouldn't that turn this from 1 proc to 2 procs?

Reply:
Yes, they appear to be two separate behaviors, it's not good to have them split into two procs.

You can move the checks and such to their own proc to share the behavior, but a huge chunk of this code is just split into an if block. We *want* those to be separate procs.